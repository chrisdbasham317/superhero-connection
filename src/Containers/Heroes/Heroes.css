.section--heroes {
  display: flex;
  flex-flow: row wrap;
  height: 80vh;
  overflow: scroll;
  padding: 0 3vw;
}