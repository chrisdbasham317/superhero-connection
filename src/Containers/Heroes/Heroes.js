import React from 'react';
import { connect } from 'react-redux';
import HeroCard from '../../Components/HeroCard/HeroCard';
import './Heroes.css';

export const Heroes = ({ heroes, isLoading }) => {
  const heroCards = heroes.map(hero => {
    return (<HeroCard
      key={hero.id}
      id={hero.id}
      name={hero.name}
      img={hero.images.md}
    />)
  })
  return (
    <section className='section--heroes'>
      {!isLoading && heroCards}
    </section>
  )
}

export const mapStateToProps = ({ heroes, isLoading }) => ({
  heroes,
  isLoading
})

export default connect(mapStateToProps)(Heroes);