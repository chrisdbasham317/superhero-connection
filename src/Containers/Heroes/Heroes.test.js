import React from 'react';
import { shallow } from 'enzyme';
import { Heroes, mapStateToProps } from './Heroes';

describe('Heroes', () => {
  let wrapper
  const mockHeroes = [
    {
      id: 4,
      name: 'superman',
      images: {
        md: 'medium image',
        lg: 'large image'
      },
      powerstats: {
        strength: 100,
        intelligence: 100,
      }
    },
    {
      id: 5,
      name: 'superboy',
      images: {
        md: 'medium image',
        lg: 'large image'
      },
      powerstats: {
        strength: 100,
        intelligence: 85,
      }
    }
  ]  

  beforeEach(() => {
    wrapper = shallow(<Heroes
      heroes={mockHeroes}
      isLoading={false}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  
})