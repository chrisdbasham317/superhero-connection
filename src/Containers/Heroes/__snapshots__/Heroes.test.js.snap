// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heroes should match the snapshot 1`] = `
<section
  className="section--heroes"
>
  <HeroCard
    id={4}
    img="medium image"
    key="4"
    name="superman"
  />
  <HeroCard
    id={5}
    img="medium image"
    key="5"
    name="superboy"
  />
</section>
`;
